### DISCUSSION TITLE
How to place boundary events in BPMN diagrams

### DISCUSSION THREAD

**Discussion by ScribbleGhost** [jgraph/drawio#3342] [Q&A]
Boundary events in BPMN are events that are attached to activities such as tasks.

I know I can drag and drop intermediate events to a task, but there is no snapping and the event doesn't move with the task if I do it.

Since BPMN is implemented in drawio, what is the right way to place a boundary event on a task?

**Comment by davidjgraph** [comment]
You're correct that boundary events in BPMN are events that are attached to activities such as tasks. To add a boundary event to a task in draw.io, you can follow these steps:

    Drag and drop the intermediate event that you want to use as the boundary event onto the canvas.

    Click on the boundary event to select it.

    Hover over the edge of the task that you want to attach the boundary event to. You should see the edge highlight in blue.

    Click and drag the boundary event onto the edge of the task. You should see a red outline appear around the task to indicate that the boundary event is attached.

    Release the mouse button to place the boundary event on the task.

If snapping isn't working as expected, you can turn it on or off by going to the View menu and selecting Snap to Guides. If it's already on and not working, you may need to adjust the snap settings in the Options menu to fine-tune the behavior.

**Reply by ScribbleGhost** [reply]
That doesn't work for me. The event never sticks to the edge of the task.

![44834673](https://user-images.githubusercontent.com/49068170/222496708-93bdd499-6c6e-498e-ab2e-030af8bc78b9.gif)

I am using the latest draw.io for desktop.

Are you able to do this on your version?

**Reply by davidjgraph** [reply]
No, we just use GPT to generate answers when threads go unanswered. We'll check.

**Comment by egrimate** [comment]
There will be a fix for this in the next release. Now border events should stick to tasks, when dropping on the as in your example.

**Reply by davidjgraph** [reply]
Fixed in 21.0.2

**Reply by alirizv** [reply]
This does not seem to be fixed, I am on V28.1.2

**Reply by davidjgraph** [reply]
Please open a new issue with full details.
